apps/worker/src/batch_announcer/ipfs.announcer.ts

```diff
@@ -1,16 +1,334 @@
 import { Injectable, Logger } from '@nestjs/common';
+import { PassThrough } from 'node:stream';
+import { ParquetWriter } from '@dsnp/parquetjs';
+import parquet from '@dsnp/frequency-schemas/parquet';
+import {
+  ActivityContentImageLink,
+  ActivityContentProfile,
+  ActivityContentTag,
+  ActivityContentAttachment,
+  ActivityContentLink,
+  ActivityContentImage,
+  ActivityContentVideoLink,
+  ActivityContentVideo,
+  ActivityContentAudio,
+  ActivityContentAudioLink,
+} from '@dsnp/activity-content/types';
+import { BlockchainService } from '../blockchain/blockchain.service';
 import { ConfigService } from '../../../api/src/config/config.service';
 import { IBatchAnnouncerJobData } from '../interfaces/batch-announcer.job.interface';
+import { IPublisherJob } from '../interfaces/publisher-job.interface';
+import { IpfsService } from '../../../../libs/common/src/utils/ipfs.client';
+import { calculateDsnpHash } from '../../../../libs/common/src/utils/ipfs';
+import {
+  AnnouncementTypeDto,
+  BroadcastDto,
+  ReplyDto,
+  ReactionDto,
+  UpdateDto,
+  ProfileDto,
+  UpdateAnnouncementTypeDto,
+  TagTypeDto,
+  AttachmentTypeDto,
+  AssetDto,
+} from '../../../../libs/common/src';
+import { createBroadcast, createReply, createReaction, createNote, createProfile } from '../../../../libs/common/src/interfaces/dsnp';
 
 @Injectable()
-export class IPFSAnnouncer {
+export class IpfsAnnouncer {
   private logger: Logger;
 
-  constructor(private configService: ConfigService) {
-    this.logger = new Logger(IPFSAnnouncer.name);
+  constructor(
+    private configService: ConfigService,
+    private blockchainService: BlockchainService,
+    private ipfsService: IpfsService,
+  ) {
+    this.logger = new Logger(IpfsAnnouncer.name);
   }
 
-  public async announce(batchJob: IBatchAnnouncerJobData): Promise<void> {
-    this.logger.log(`Announcing batch ${batchJob.batchId} on IPFS`);
+  public async announce(batchJob: IBatchAnnouncerJobData): Promise<IPublisherJob> {
+    this.logger.debug(`Announcing batch ${batchJob.batchId} on IPFS`);
+    const { batchId, schemaId, announcements } = batchJob;
+
+    const frequencySchema = await this.blockchainService.getSchema(schemaId);
+    const schema = JSON.parse(frequencySchema.model.toString());
+    if (!schema) {
+      throw new Error(`Unable to parse schema for schemaId ${schemaId}`);
+    }
+
+    const [parquetSchema, writerOptions] = parquet.fromFrequencySchema(schema);
+    const publishStream = new PassThrough();
+
+    const writer = await ParquetWriter.openStream(parquetSchema, publishStream as any, writerOptions);
+
+    announcements.forEach(async (announcement) => {
+      switch (announcement.announcementType) {
+        case AnnouncementTypeDto.BROADCAST:
+          await this.processBroadcast(writer, announcement.content as BroadcastDto, announcement.dsnpUserId);
```

Comment:
all of announcement creation will be done in request processor and in here you should already have everything needed for individual announcements